cmd/utils/flags.go

```diff
@@ -1769,7 +1770,24 @@ func RegisterEthService(stack *node.Node, cfg *ethconfig.Config) (ethapi.Backend
 		stack.RegisterAPIs(tracers.APIs(backend.ApiBackend))
 		return backend.ApiBackend, nil
 	}
-	backend, err := eth.New(stack, cfg)
+
+	// initialize L1 client for sync service
+	// note: we need to do this here to avoid circular dependency
+	l1EndpointUrl := stack.Config().L1Endpoint
+	var l1Client *ethclient.Client = nil
```

Comment:
`var l1Client *ethclient.Client = nil ` => `var l1Client *ethclient.Client` 
this will be better.